src/OSPSuite.Infrastructure/Services/HistoryTask.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Data.Common;
+using OSPSuite.Assets;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Core.Events;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Core.Services;
+using OSPSuite.Infrastructure.Serialization.ORM.MetaData;
+using OSPSuite.Utility.Events;
+
+namespace OSPSuite.Infrastructure.Services
+{
+   public class HistoryTask: IHistoryTask
+   {
+      private readonly IProjectRetriever _projectRetriever;
+      private readonly IHistoryManagerRetriever _historyManagerRetriever;
+      private readonly IDialogCreator _dialogCreator;
+      private readonly SQLiteProjectCommandExecuter _commandExecuter;
+      private readonly IEventPublisher _eventPublisher;
+      private readonly ICommandMetaDataRepository _commandMetaDataRepository;
+
+      public HistoryTask(
+         IProjectRetriever projectRetriever, 
+         IHistoryManagerRetriever historyManagerRetriever, 
+         IDialogCreator dialogCreator, 
+         SQLiteProjectCommandExecuter commandExecuter,
+         IEventPublisher eventPublisher,
+         ICommandMetaDataRepository commandMetaDataRepository
+         )
+
+      {
+         _projectRetriever = projectRetriever;
+         _historyManagerRetriever = historyManagerRetriever;
+         _dialogCreator = dialogCreator;
+         _commandExecuter = commandExecuter;
+         _eventPublisher = eventPublisher;
+         _commandMetaDataRepository = commandMetaDataRepository;
+      }
+
+      public void ClearHistory()
+      {
+         var clear = _dialogCreator.MessageBoxYesNo(Captions.ReallyClearHistory, Captions.Clear, Captions.CancelButton);
+         if (clear == ViewResult.No) 
+            return;
+
+         clearProjectHistory();
+         clearCurrentHistory();
+      }
+
+      private void clearCurrentHistory()
+      {
+         var historyManager = _historyManagerRetriever.Current;
```

Comment:
This should clear all new commands created so far.
I will need to test all of that in PKSim to see if something is not working as expected